core/shared/src/main/scala/org/http4s/Credentials.scala

```diff
@@ -102,21 +102,29 @@ object BasicCredentials {
   private def decode(bytes: Array[Byte], charset: JavaCharset): String =
     new String(bytes, charset)
 
-  def apply(token: String): BasicCredentials = {
-    val bytes = Base64.getDecoder.decode(token)
-    val (userPass, charset) = decode(bytes, utf8CharsetDecoder)
-      .fold(_ => (decode(bytes, fallbackCharset), fallbackCharset), up => (up, utf8Charset))
-    userPass.indexOf(':') match {
-      case -1 => apply(userPass, "", charset)
-      case ix => apply(userPass.substring(0, ix), userPass.substring(ix + 1), charset)
+  private def fromString0(token: String): Try[BasicCredentials] =
+    Try(Base64.getDecoder.decode(token)).map { bytes =>
+      val (userPass, charset) = decode(bytes, utf8CharsetDecoder)
+        .fold(_ => (decode(bytes, fallbackCharset), fallbackCharset), up => (up, utf8Charset))
+      userPass.indexOf(':') match {
+        case -1 => apply(userPass, "", charset)
+        case ix => apply(userPass.substring(0, ix), userPass.substring(ix + 1), charset)
+      }
     }
-  }
+
+  @deprecated("Use fromString instead", "0.23.24")
```

Comment:
I know that I have annoyed you numerous times with this, but shouldn't we have a version related to 1.0?

Reply:
I can't remember what we decided before 😂 

Anyway we need to hold this PR for now, somehow it broke the website.

Reply:
I thought about it again. Suppose the user wants to check the release notes for more information about the deprecation, which release notes should they check? Because the deprecation was in 0.23.24, they need to check the release notes for 0.23.24. If they check the release notes for 1.0.0-M41 all they will see is this big merge PR.